crates/starknet-devnet/tests/test_simulate_transactions.rs

```diff
@@ -232,6 +234,204 @@ mod simulation_tests {
         );
     }
 
+    async fn update_gas_scenario(devnet: BackgroundDevnet, assert_chain_id: &str) {
+        // get account
+        let (signer, account_address) = devnet.get_first_predeployed_account().await;
+        let account = SingleOwnerAccount::new(
+            devnet.clone_provider(),
+            signer,
+            account_address,
+            FieldElement::from_hex_be(assert_chain_id).unwrap(),
+            ExecutionEncoding::New,
+        );
+
+        // get class
+        let (flattened_contract_artifact, casm_hash) =
+            get_flattened_sierra_contract_and_casm_hash(CAIRO_1_CONTRACT_PATH);
+
+        let max_fee = FieldElement::ZERO;
+        let nonce = FieldElement::from_hex_be("0x0").unwrap();
+
+        let signature = account
+            .declare(Arc::new(flattened_contract_artifact.clone()), casm_hash)
+            .max_fee(max_fee)
+            .nonce(nonce)
+            .prepared()
+            .unwrap()
+            .get_declare_request(false)
+            .await
+            .unwrap()
+            .signature;
+
+        let signature_hex: Vec<String> = iter_to_hex_felt(&signature);
+        let sender_address_hex = to_hex_felt(&account_address);
+        let get_params = |simulation_flags: &[&str]| -> serde_json::Value {
+            json!({
+                "block_id": "latest",
+                "simulation_flags": simulation_flags,
+                "transactions": [
+                    {
+                        "type": "DECLARE",
+                        "sender_address": sender_address_hex,
+                        "compiled_class_hash": to_hex_felt(&casm_hash),
+                        "max_fee": to_hex_felt(&max_fee),
+                        "version": "0x2",
+                        "signature": signature_hex,
+                        "nonce": to_num_as_hex(&nonce),
+                        "contract_class": flattened_contract_artifact,
+                    }
+                ]
+            })
+        };
+
+        let chain_id = &devnet.send_custom_rpc("starknet_chainId", json!({})).await.unwrap();
+        assert_eq!(chain_id, assert_chain_id);
```

Comment:
Why is this assertion done?

Reply:
chain id might be wrong since we need to reinitialized `Starknet::init_block_context` so this is why I added this extra check based on my knowledge about implementation.